Review of main readme and model run (#26)
nestauk/cci_cameroon: cci_cameroon/pipeline/model_workflow/README.md

# Running the models

<br>

> <br>

Reviewer: ```suggestion
```

> The x2 scripts in this folder can be used to run the classification and clustering models on new rumours (defaulted to the test set). The scripts use the held out test dataset to first classfiy the rumours and then cluster the rumours that cannot be classified. Both scripts can be run seperately and can use any new rumour datasets that are in the correct format (contain a comment field for predicting and clustering).

Reviewer: ```suggestion
The scripts in this folder can be used to run the classification and clustering models on new rumours (in this case, represented by the test data). The first classfiy the rumours and then cluster the ones that cannot be classified. Both scripts can be run seperately and can use any new rumour datasets that are in the correct format (contain a comment field for predicting and clustering).
```
Might be a good idea to describe the format in a bit more detail - is it a csv, json, xlsx? Does it need to have a specific name etc.?

### Steps to take before running

To run the models you will first need to setup the project. Follow the below two steps to do this:

1. Clone the project and `cd` into the `cci_cameroon` directory
2. Run the command `make install` to create the virtual environment and install dependencies
3. Inside the project directory run `make inputs-pull` to access the data from S3 (for those with access to the Nesta S3 account)

> To note the project is setup using the Nesta Cookiecutter (guidelines on the Nesta Cookiecutter [can be found here](https://nestauk.github.io/ds-cookiecutter/structure/)).

Reviewer: I'm not sure that it's obvious a user/reader should come here to understand how to set up the repo.

#### Pre-trained models (classification)

> The pre-trained classification model is saved to outputs/model. This model is fitted using the same IFRC dataset that the test set came from. The script to run the file is in the pipeline folder and is called 'classification_model_save\*.'

Reviewer: ```suggestion
The pre-trained classification model is saved to `outputs/model`. This model is fitted using the same IFRC dataset that the test set came from. The script to run the file is in the pipeline folder and is called `classification_model_save`\*.
```

The MultiLabelBinarizer model is used to access the classes to create the predictions dataset. This model is also created and saved to outputs by running `classification_model_save.py`.

> \*This script relies on the training set being saved in `outputs/data/data_for_modelling/`. See the main readme page for more information.

Reviewer: Normally I would say that if you're going to reference other readmes you should link to them and the appropriate section. However, given my comment about putting an outline on the main readme and moving the detail here, I would stick with that and delete the last sentence here.

#### Input needed

After you setup the project you will need your test dataset. To build our models we used the dataset provided to us by the IFRC and Cameroon Red Cross containing rumours, beliefs and observations from community members (more information on the data source can be found on the [IFRC.Go website](https://go.ifrc.org/emergencies/4583#community-data)). To run the scripts the data needs to be in the following format:

- Filename: `X_test.xlsx`

> - Saved in `inputs/data/data_for_modelling/`

Reviewer: I'm not expecting any big changes to the codebase at this point, but in future it might be worth considering turning something like this into a CLI tool with something like [click](https://click.palletsprojects.com/en/8.1.x/).

| id      | comment     |
| ------- | ----------- |
| numeric | rumour text |

> The reason for having an ID column is so you can reference back the prediction files to the original test dataset (if needed for reporting results).

Reviewer: ```suggestion
The reason for having an ID column is so you can reference the prediction files back to the original test dataset (if needed for reporting results).
```

### Run the models

Perform the following steps to run the models:

- run `python3 classification_model_run.py`

> - run `python3 clustering_model_run.py`

Reviewer: ```suggestion
- run `python classification_model_run.py`
- run `python clustering_model_run.py`
```
